### DISCUSSION TITLE
Inconsistent Issue Timeline Data Retrieval with and without Authorization

### DISCUSSION THREAD

**Discussion by Momo-Not-Emo** [community/community#107627] [Apps, API and Webhooks]
### Select Topic Area

Bug

#### Issue Description:
When making requests to retrieve the issue timeline using the GitHub REST API, I have observed different data being returned with and without Authorization. Surprisingly, the data requested without Authorization is more complete and aligns with my expectations.

#### Steps to Reproduce:
The issue can be easily reproduced by running the following commands:

Without Authorization:
```sh
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/soot-oss/soot/issues/1577/timeline
```

With Authorization:
```
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer MY_TOKEN" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/soot-oss/soot/issues/1577/timeline
```

##### Observations:
In the first case (without Authorization), the data received is more complete and aligned with my expectations for the issue timeline. However, when the request is made with Authorization, some data appears to be missing.

- Response with Authorization (screenshot of Postman): The returned list only contains one element
    <img width="495" alt="image" src="https://github.com/community/community/assets/149540451/fbc6c88d-d1b4-404c-b4bf-01e4c60fb047">

- Response without Authorization: The returned list contains (1) four elements and (2) the element key set is different from the above response. 
    <img width="495" alt="image" src="https://github.com/community/community/assets/149540451/0dc3fb82-cff0-48ff-9ca8-b2a7626d2304">

#### Additional Information:
I have also observed an increase in the "X-RateLimit-Limit" header in the response, indicating that my authorization method is functioning correctly. Therefore, I believe the issue is not related to the authentication method but rather to the data retrieval process.

**Comment by afifaniks** [comment]
Did you find any fix? Facing the same issue.

**Comment by afifaniks** [comment]
@Momo-Not-Emo I think I found a workaround. It seems the issue lies with the type of personal access token being used. When using a "Fine-grained token," inconsistencies arise compared to the version without any tokens. Therefore, I proceeded to use a **"classic token (ghp_*)"** with all permissions enabled (although this step might not have been necessary), just to ensure functionality, and it worked just fine!!
